internal/castai/castai.go

```diff
@@ -86,6 +88,10 @@ func NewDefaultDeltaHTTPClient() *http.Client {
 }
 
 func createHTTPTransport() *http.Transport {
+	tlsConfig, err := createTLSConfig()
+	if err != nil {
+		panic(fmt.Errorf("creating TLS config: %v", err))
```

Comment:
Can we do a graceful shutdown and return the error to the `main()` function instead of panicing? Returning an error will execute all the `defer`ed statements and enrich the error with a trace.

Reply:
Removed panic and error is returned to `main()` now.